src/app/help/help-search/help-search.component.ts

```diff
@@ -284,34 +284,31 @@ export class HelpSearchComponent implements OnInit {
    */
   public onExport(): void {
     this.displayExportDialog = true
+    this.assignedProductNames = Array.from(new Set(this.resultsForDisplay.map((item) => item.productDisplayName!)))
   }
   public onExportConfirmation(): void {
-    if (this.selectedResults && this.selectedResults.length > 0) {
-      this.selectedProductNames = this.selectedResults.map((item) => item.productName!)
-      this.helpInternalAPIService
-        .exportHelps({ exportHelpsRequest: { productNames: this.selectedProductNames } })
-        .subscribe({
-          next: (item) => {
-            const helpsJson = JSON.stringify(item, null, 2)
-            FileSaver.saveAs(
-              new Blob([helpsJson], { type: 'text/json' }),
-              'onecx-help-items_' + this.getCurrentDateTime() + '.json'
-            )
-            this.msgService.success({ summaryKey: 'ACTIONS.EXPORT.MESSAGE.HELP_ITEM.EXPORT_OK' })
-            this.displayExportDialog = false
-            this.selectedResults = []
-            this.selectedProductNames = []
-          },
-          error: (err) => {
-            this.msgService.error({ summaryKey: 'ACTIONS.EXPORT.MESSAGE.HELP_ITEM.EXPORT_NOK' })
-            console.error(err)
-          }
-        })
+    if (this.selectedProductNames && this.selectedProductNames.length > 0) {
```

Comment:
kann man das nicht auch vereinfachen zu
this.selectedProductNames?.length > 0

Reply:
Vereinfacht